site/content/user-guide/java_client_sender.md

```diff
@@ -0,0 +1,268 @@
+
+This allows experienced developers to just read (and understand) the source code and copy it to their own implementation to get Hono connected as quickly as possible.
+
+Here is the code:
+
+```java
+package org.eclipse.hono;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.proton.ProtonClientOptions;
+import org.eclipse.hono.client.HonoClient;
+import org.eclipse.hono.client.MessageSender;
+import org.eclipse.hono.connection.ConnectionFactoryImpl;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.stream.IntStream;
+
+
+public class App {
+    public static final String HONO_HOST = "localhost";
+    public static final short HONO_PORT = 5672;
+
+    public static final String TENANT_ID = "DEFAULT_TENANT";
+    public static final String DEVICE_ID = "4711";
+
+    public static final int COUNT = 50;
+
+    private final Vertx vertx = Vertx.vertx();
+    private final HonoClient honoUpstreamClient;
+    
+    private final CountDownLatch latch;
+
+    public App() {
+        honoUpstreamClient = new HonoClient(vertx,
+                ConnectionFactoryImpl.ConnectionFactoryBuilder.newBuilder()
+                        .vertx(vertx)
+                        .host(HONO_HOST)
+                        .port(HONO_PORT)
+                        .user("hono-client")
+                        .password("secret")
+                        .trustStorePath("certs/trusted-certs.pem")
+                        .disableHostnameVerification()
+                        .build());
+        latch = new CountDownLatch(1);
+    }
+
+    public static void main(String[] args) throws Exception {
+        System.out.println("Starting upstream sender...");
+        App app = new App();
+        app.sendUpstreamTelemetryData();
+        System.out.println("Finishing upstream sender.");
+    }
+
+
+    private void sendUpstreamTelemetryData() throws Exception {
+        Context ctx = vertx.getOrCreateContext();
+
+        final Future<MessageSender> senderFuture = Future.future();
+
+        senderFuture.setHandler(result -> {
+            if (result.succeeded()) {
+                latch.countDown();
+                return;
+            }
+            System.err.println("honoUpstreamClient could not create telemetry sender!");
+        });
+
+        ctx.runOnContext(go -> {
+            final Future<HonoClient> connectionTracker = Future.future();
+            honoUpstreamClient.connect(new ProtonClientOptions(), connectionTracker.completer());
+
+            connectionTracker.compose(honoClient -> {
+                        honoClient.getOrCreateTelemetrySender(TENANT_ID, senderFuture.completer());
+                    },
+                    senderFuture);
+        });
+
+        latch.await();
```

Comment:
After fix #106 : the senderFuture is called with a failed result and the program terminates with an error message regularly.